File: examples/nodejs/oauth.js
const { Config, OAuth } = require("longport");

async function main() {
  const oauth = new OAuth("your-client-id");
  const token = await oauth.authorize((url) => {
    console.log("Open this URL to authorize:", url);
  });
  console.log("Access token:", token.accessToken);
  console.log("Expires at:", token.expiresAt);

  const config = Config.fromOauth("your-client-id", token.accessToken);
  console.log("Config created:", config);
}

Promise.all([main()]).catch((err) => console.error(err));

File: examples/python/oauth.py
import asyncio
from longport.openapi import Config, OAuth


async def main():
    oauth = OAuth("your-client-id")
    token = await oauth.authorize(lambda url: print(f"Open this URL to authorize: {url}"))
    print(f"Access token: {token.access_token}")
    print(f"Expires at: {token.expires_at}")

    config = Config.from_oauth("your-client-id", token.access_token)
    print(f"Config created: {config}")


asyncio.run(main())
